static-assets/scripts/main.js

```diff
@@ -716,9 +720,113 @@
     }
   ]);
 
+  app.service('passwordRequirements', [
+    '$rootScope', '$http', 'Constants', '$cookies', '$timeout', '$window', '$translate',
+    function ($rootScope, $http, Constants, $cookies, $timeout, $window, $translate) {
+      var me = this,
+        generalRegExpre = passwordRequirementsRegex,
+        generalRegExpreWithoutGroups = generalRegExpre.replace(/\?<(.*?)>/g, ""),
+        captureGroup = generalRegExpre.match(/\(\?<.*?>.*?\)/g),
+        messages = {
+          hasNumbers: formatMessage(passwordRequirementMessages.hasNumbers),
+          hasLowercase: formatMessage(passwordRequirementMessages.hasLowercase),
+          hasUppercase: formatMessage(passwordRequirementMessages.hasUppercase),
+          hasSpecialChars: formatMessage(passwordRequirementMessages.hasSpecialChars),
+          noSpaces: formatMessage(passwordRequirementMessages.noSpaces),
+          minLength: formatMessage(passwordRequirementMessages.minLength),
+          passwordValidation: formatMessage(passwordRequirementMessages.passwordValidation),
+          validPassword: formatMessage(passwordRequirementMessages.validPassword),
+          invalidPassword: formatMessage(passwordRequirementMessages.invalidPassword)
+        };
+
+      this.init = function ($scope, elt) {
+        try {
+          let isRegExpValid = new RegExp(generalRegExpre);
+          if (isRegExpValid && captureGroup.length > 0) {
+            this.runValidation($scope, elt);
+          }
+        } catch (error) {
+          try {
+            let isRegExpValid = new RegExp(generalRegExpreWithoutGroups);
+            if (isRegExpValid) {
+              this.runValidation($scope, elt, true);
+            }
+          } catch (error) {
+            console.log("Default Password Validation");
+          }
+        }
+      }
+
+      this.creatingPassValHTML = function (content, staticTemplate) {
+        var html = '<div class="password-popover">';
+        var validPass = false;
+        var isGeneralRegExpWithoutGroupsValid = content ? content.match(generalRegExpreWithoutGroups) : false;
+        captureGroup.forEach(captureGroup => {
+          let captureGroupName = captureGroup.match(/\?<(.*?)>/g);
+          if (staticTemplate) {
+            html += '<ul class="static"><li>'
+          } else {
+            let isValid = content ? content.match(captureGroup) : false;
+            html += '<ul><li>'
+            if (isValid) {
+              html += '<span class="list-icon fa fa-check-circle green"></span>';
+            } else {
+              html += '<span class="list-icon fa fa-times-circle red "></span>';
+              validPass = true;
+            }
+          }
+          html += messages[captureGroupName[0].replace(/\?<|>/g, "")] +
+            '</li>';
+        });
+        html += '</ul>';
+        if (staticTemplate) {
```

Comment:
single if with dual. condittion

Reply:
Else is only for the second condition. It doesn't depend on first if (condition)